frontend/database/modules/datasets.js

```diff
@@ -597,6 +596,65 @@ const actions = {
     });
     await _refreshDatasetAggregations({ dataset: paginatedDataset });
   },
+
+  async getRules(_,{ dataset }) {
+    const { response } = await ObservationDataset.api().get(
+      `/datasets/${dataset.task}/${dataset.name}/labeling/rules`,
+    );
+    return response.data;
+  },
+
+  async defineRule(_,{ dataset, label }) {
+    await ObservationDataset.api().post(
+      `/datasets/${dataset.task}/${dataset.name}/labeling/rules`,
+      {
+        query: dataset.query.text,
+        label: label,
+        description: dataset.query.text
+      }
+    );
+  },
+
+  async updateRule(_,{ dataset, label }) {
+    await ObservationDataset.api().patch(
+      `/datasets/${dataset.task}/${dataset.name}/labeling/rules/${dataset.query.text}`,
+      {
+        label: label,
+        description: dataset.query.text
+      }
+    );
+  },
+
+  async getRule(_,{ dataset, query}) {
+    try {
+      const { response } = await ObservationDataset.api().get(
+        `/datasets/${dataset.task}/${dataset.name}/labeling/rules/${query}`,
+      )
+      return response.data;
+    } catch {
+      return undefined;
+    }
+  },
+
+  async deleteRule(_,{ dataset, query }) {
+    await ObservationDataset.api().delete(
+      `/datasets/${dataset.task}/${dataset.name}/labeling/rules/${query}`,
+    );
+  },
+
+  async getRulesMetrics(_,{ dataset }) {
+    const { response } = await ObservationDataset.api().get(
+      `/datasets/${dataset.task}/${dataset.name}/labeling/rules/metrics`,
+    );
+    return response.data;
+  },
+
+  async getRuleMetricsByLabel(_,{ dataset, query, label }) {
+    const { response } = await ObservationDataset.api().get(
+      `/datasets/${dataset.task}/${dataset.name}/labeling/rules/${query}/metrics?label=${label}`,
+    );
+    return response.data;
+  }
```

Comment:
All this actions should be defined only for text classification. 

Maybe better move  to dataset/modules/text_classification